addons/survey/models/survey_user.py

```diff
@@ -95,6 +123,24 @@ def action_print_answers(self):
             'url': '/survey/print/%s?answer_token=%s' % (self.survey_id.access_token, self.token)
         }
 
+    @api.multi
+    def _is_time_limit_reached(self):
+        """ Checks that this user_input is not exceeding the survey's time limit. """
+        self.ensure_one()
+
+        return self.survey_id.is_time_limited \
+            and fields.Datetime.now() > (self.start_datetime + relativedelta(minutes=self.survey_id.time_limit))
+
+    @api.multi
+    def _send_certification(self):
+        """ Will send the certification email with attached document if
+        - The survey is a certification
+        - It has a certification_email_template_id set
+        - The user succeeded the test """
+        for user_input in self:
+            if user_input.survey_id.certificate and user_input.quizz_passed and user_input.survey_id.certification_email_template_id:
+                user_input.survey_id.certification_email_template_id.send_mail(user_input.id, force_send=True, notif_layout="mail.mail_notification_light")
```

Comment:
Is the force_send really necessary ? I mean using the mail queue should be sufficient, isn't it ?

Reply:
I removed it after all.

When we have the "download certif" button the flow will be nice. (Now the user will have to wait for the cron to trigger and he does not even know he's going to receive an email... But it's temporary.)